### PR TITLE
Added SecurityContextConstraints package

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/scc/scc.go [hunk 1]
```diff
@@ -0,0 +1,385 @@
+package scc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	v1 "github.com/openshift/api/security/v1"
```
Comment (evgenLevin) [inline pkg/scc/scc.go:10]: securityV1

File: pkg/scc/scc.go [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+)
+
+// Builder provides struct for SecurityContextConstraints object containing connection
+// to the cluster SecurityContextConstraints definition.
+type Builder struct {
+	// SecurityContextConstraints definition. Used to create SecurityContextConstraints object
+	Definition *v1.SecurityContextConstraints
+	// Created SecurityContextConstraints object
+	Object *v1.SecurityContextConstraints
+
```
Comment (evgenLevin) [inline pkg/scc/scc.go:23]: remove space

File: pkg/scc/scc.go [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+		return builder
+	}
+
+	builder.Definition.AllowedCapabilities = append(builder.Definition.AllowedCapabilities, allowCapabilities...)
+
+	return builder
+}
+
+// WithFSGroup adds list of allow capabilities to SecurityContextConstraints.
```
Comment (evgenLevin) [inline pkg/scc/scc.go:188]: Fix the comment please

File: pkg/scc/scc.go [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+	return builder, err
+}
+
+// Delete removes a SecurityContextConstraints.
+func (builder *Builder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Removing SecurityContextConstraints %ss", builder.Definition.Name)
```
Comment (evgenLevin) [inline pkg/scc/scc.go:311]: typo %ss

**Review by evgenLevin — COMMENTED** [review]
few comments

File: pkg/scc/scc.go [hunk 5]
```diff
@@ -0,0 +1,384 @@
...
+
+	return builder
+}
+
+// WithFSGroup adds fsGroup to SecurityContextConstraints.
+func (builder *Builder) WithFSGroup(fsGroup string) *Builder {
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		"fsGroup: %s", builder.Definition.Name, fsGroup)
+
+	if valid, _ := builder.validate(); !valid {
```
Comment (trewest) [inline pkg/scc/scc.go:192]: Need this block before glog

File: pkg/scc/scc.go [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+
+	return builder
+}
+
+// WithSeccompProfiles adds list of seccompProfiles to SecurityContextConstraints.
+func (builder *Builder) WithSeccompProfiles(seccompProfiles []string) *Builder {
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		"SeccompProfiles: %v", builder.Definition.Name, seccompProfiles)
+
+	if valid, _ := builder.validate(); !valid {
```
Comment (trewest) [inline pkg/scc/scc.go:214]: Need this block before glog

File: pkg/scc/scc.go [hunk 7]
```diff
@@ -0,0 +1,384 @@
...
+
+	return builder
+}
+
+// WithSupplementalGroups adds SupplementalGroups to SecurityContextConstraints.
+func (builder *Builder) WithSupplementalGroups(supplementalGroupsType string) *Builder {
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		"supplementalGroupsType: %s", builder.Definition.Name, supplementalGroupsType)
+
+	if valid, _ := builder.validate(); !valid {
```
Comment (trewest) [inline pkg/scc/scc.go:242]: Need this block before glog

File: pkg/scc/scc.go [hunk 8]
```diff
@@ -0,0 +1,384 @@
...
+
+	return builder
+}
+
+// WithUsers adds users to SecurityContextConstraints.
+func (builder *Builder) WithUsers(users []string) *Builder {
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		"users: %v", builder.Definition.Name, users)
+
+	if valid, _ := builder.validate(); !valid {
```
Comment (trewest) [inline pkg/scc/scc.go:264]: Need this block before glog

File: pkg/scc/scc.go [hunk 9]
```diff
@@ -0,0 +1,384 @@
...
+// WithPrivilegedEscalation adds bool flag to the allowPrivilegeEscalation of SecurityContextConstraints.
+func (builder *Builder) WithPrivilegedEscalation(allowPrivilegedEscalation bool) *Builder {
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		" allowPrivilegedEscalation: %t flag", builder.Definition.Name, allowPrivilegedEscalation)
+
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	builder.Definition.AllowPrivilegedContainer = allowPrivilegedEscalation
```
Comment (trewest) [inline pkg/scc/scc.go:125]: This is the same as the previous method, just with a different parameter name

**Review by trewest — COMMENTED** [review]
Just a few comments

File: pkg/scc/scc.go [hunk 10]
```diff
@@ -0,0 +1,384 @@
...
+		glog.V(100).Infof("The selinuxContext of the SecurityContextConstraints is empty")
+
+		builder.errorMsg = "SecurityContextConstraints 'selinuxContext' cannot be empty"
+	}
+
+	return &builder
+}
+
+// PullSecurityContextConstraints pulls existing SecurityContextConstraints from cluster.
+func PullSecurityContextConstraints(apiClient *clients.Settings, name string) (*Builder, error) {
```
[the remark this answers is not in the document]
Comment (kononovn) [reply pkg/scc/scc.go:75]: +1. fixed

File: pkg/scc/scc.go (hunk 9 above)
Comment (kononovn) [reply pkg/scc/scc.go:125]: +1. Fixed

File: pkg/scc/scc.go [hunk 11]
```diff
@@ -0,0 +1,384 @@
...
+
+	return &builder, nil
+}
+
+// WithPrivilegedContainer adds bool flag to the allowPrivilegedContainer of SecurityContextConstraints.
+func (builder *Builder) WithPrivilegedContainer(allowPrivileged bool) *Builder {
+	glog.V(100).Infof("Redefining SecurityContextConstraints %s with"+
+		" AllowPrivilegedContainer: %t flag", builder.Definition.Name, allowPrivileged)
+
+	if valid, _ := builder.validate(); !valid {
```
[the remark this answers is not in the document]
Comment (trewest) [reply pkg/scc/scc.go:107]: @kononovn still need to move this one

**Review by trewest — COMMENTED** [review]
/lgtm

**Pull request merged**